src/Command/CheckAllSchemasAreValidAvroCommand.php

```diff
@@ -80,4 +90,115 @@ private function checkSchemas(array $avroFiles, array &$failed = []): bool
 
         return 0 === count($failed);
     }
+
+    /**
+     * @param string $localSchema
+     * @throws AvroSchemaParseException
+     */
+    private function checkDefaultType(string $localSchema): void
+    {
+        $decodedSchema = json_decode($localSchema);
+        if (!property_exists($decodedSchema, 'fields')) {
+            return;
+        }
+
+        $result = [
+            'found' => 0,
+            'default' => 0,
+        ];
+
+        $result = $this->checkAllFields($decodedSchema->fields, $result);
+
+        if ($result['found'] !== $result['default']) {
+            throw new AvroSchemaParseException('Type of default value is not in field type.');
+        }
+    }
+
+    /**
+     * @param array<string, mixed> $schemaFields
+     * @param array<string, mixed> $result
+     * @return array<string, mixed>
+     * @throws AvroSchemaParseException
+     */
+    private function checkAllFields(array $schemaFields, array $result): array
+    {
+        foreach ($schemaFields as $field) {
+            $fieldTypes = $field->type;
+            if (property_exists($field, 'default')) {
+                $result['default']++;
+            }
+
+            if (is_array($fieldTypes)) {
+                foreach ($fieldTypes as $fieldType) {
+                    $result = $this->checkSingleField($fieldType, $field, $result);
+                }
+            }
+
+            if (!is_array($fieldTypes)) {
+                $result = $this->checkSingleField($fieldTypes, $field, $result);
+            }
+        }
+
+        return $result;
+    }
+
+    /**
+     * @param mixed $fieldType
+     * @param mixed $field
+     * @param array<string, mixed> $result
+     * @return array<string, mixed>
+     * @throws AvroSchemaParseException
+     */
+    private function checkSingleField($fieldType, $field, array $result): array
+    {
+        $defaultType = null;
+
+        if (property_exists($field, 'default')) {
+            $defaultType = strtolower(gettype($field->default));
+
+            if (is_string($fieldType)) {
+                if (
+                    self::TYPE_MAP[$defaultType] === $fieldType
+                    || $this->isContainedInBiggerType(self::TYPE_MAP[$defaultType], $fieldType)
+                ) {
+                    $result['found']++;
+                }
+            }
+        }
+
+        if (property_exists($fieldType, 'type')) {
```

Comment:
if it is easier to check the templates instead of the whole schema, you can also do this, templates are not nested :smile: 
instead of just tracking found and default, we should "track" more so that in the end we can say, which fields failed and not just one message that it somewhere failed :v:

Reply:
Separate check for templates is added and it is now showing field names on error output.